web-portal/backend/src/utils/utils.service.ts

```diff
@@ -168,39 +169,64 @@ export class UtilsService {
 
   @Cron(CronExpression.EVERY_MINUTE)
   async watchEvent() {
-    console.log('Getting Event Logs');
-    const blockNumber = await viemClient.getBlockNumber();
-    const logs = await viemClient.getLogs({
-      event,
-      address: portrAddress,
-      fromBlock: blockNumber - BigInt(1000),
-      toBlock: blockNumber,
-    });
+    console.log('Getting Event Logs from all clients');
+
+    // Define clients and their respective names
+    const clients = [
+      { client: opClient, name: 'optimism' },
+      { client: gnosisClient, name: 'gnosis' },
+      { client: baseClient, name: 'base' }
```

Comment:
Is this manageable going forward. At what point do we have too many chains to support with the code this way?